src/tiled/stampbrush.cpp

```diff
@@ -250,6 +250,17 @@ void StampBrush::endCapture()
 
     // Intersect with the layer and translate to layer coordinates
     QRect captured = capturedArea();
+
+    captured &= QRect(tileLayer->x(), tileLayer->y(),
+                      tileLayer->width(), tileLayer->height());
+
+    //gets if the relative stagger should be the same as the base layer
+    int stagIndexOffSet;
```

Comment:
One more comment: there is no point in shortening "stagger" here, so just call it `staggerIndexOffset`.